crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -330,15 +357,17 @@ impl CheckpointContents {
     }
 
     pub fn into_inner(self) -> Vec<ExecutionDigests> {
-        self.transactions
+        self.into_v1().transactions
     }
 
     pub fn size(&self) -> usize {
-        self.transactions.len()
+        self.as_v1().transactions.len()
     }
 
-    pub fn digest(&self) -> CheckpointContentsDigest {
-        CheckpointContentsDigest::new(sha3_hash(self))
+    pub fn digest(&self) -> &CheckpointContentsDigest {
+        self.as_v1()
+            .digest
+            .get_or_init(|| CheckpointContentsDigest::new(sha3_hash(self)))
```

Comment:
How are you distinguishing between V1 and some future V2? if you're calculating the digest without including the variant byte used to distinguish between two versions of the type then i don't think we'll actually have an easy path forward for evolving this, unless this is being handled some other way.

Reply:
oh good catch, need to compute the hash on the outer structure

Reply:
wait, no, this i fine - the hash is over `self`, even though it is stored in V1, so it does include the variant tag.

Reply:
Oh...the as_v1 threw me off...